### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by NelsonVides**
These are all changes specially to make hexdocs (and everything else that uses hexdocs for that matter!) prettier. And also to make code more understandable in general.

Note btw that the license doesn't need to go onto every single file for Apache, see:
https://www.apache.org/legal/apply-license.html#copy-per-file

### CODE REVIEW

File: src/amoc_controller.erl [hunk 1]
```diff
@@ -24,17 +26,21 @@
...
-                       disabled. %% amoc_controller is disabled for the master node
-
+                       disabled.
+%% Status of the node, note that amoc_controller is disabled for the master node
 -type user_count() :: non_neg_integer().
+%% Number of users currently running in the node
 -type last_user_id() :: non_neg_integer().
+%% Highest user id registered in the node
 -type interarrival() :: non_neg_integer().
+%% Time to wait in between spawning new users
```
Comment (pawlooss1) [inline src/amoc_controller.erl:43]: These comments are bellow the types they describe. To keep it consistent with e.g. functions documentation I'd place them above instead.

File: src/amoc_distribution/amoc_dist.erl [hunk 2]
```diff
@@ -29,28 +30,42 @@ do(Scenario, Count, Settings) ->
         {Error, _} -> Error
     end.
 
+%% @doc dynamically add more users to a currently running scenario, in all nodes
+%% @see add/3
```
Comment (pawlooss1) [inline src/amoc_distribution/amoc_dist.erl:34]: I think you meant `add/2`.

File: src/amoc.erl [hunk 3]
```diff
@@ -32,6 +30,7 @@ do(Scenario, Count, Settings) ->
         Error -> Error
     end.
 
+%% @doc dynamically add more users to a currently running scenario
```
Comment (pawlooss1) [inline src/amoc.erl:33]: Some of the docs are starting with a capital letter and some don't. Please make it consistent for all modules. I think capital letter is nicer.

File: src/amoc.erl [hunk 4]
```diff
@@ -1,7 +1,8 @@
-%%==============================================================================
-%% Copyright 2023 Erlang Solutions Ltd.
-%% Licensed under the Apache License, Version 2.0 (see LICENSE file)
-%%==============================================================================
+%% @copyright 2023 Erlang Solutions Ltd.
+%% @doc API module for running locally, in a non-distributed environment
```
Comment (pawlooss1) [inline src/amoc.erl:2]: Maybe we could use `@moduledoc` attribute? https://hexdocs.pm/elixir/1.12/writing-documentation.html#module-attributes If you do so, then please change it everywhere.

File: src/amoc_throttle/amoc_throttle.erl [hunk 5]
```diff
@@ -18,26 +16,34 @@
...
--define(DEFAULT_INTERVAL, 60000).%% one minute
+-define(DEFAULT_NO_PROCESSES, 10).
+-define(DEFAULT_INTERVAL, 60000). %% one minute
+
 -type name() :: atom().
+-type rate() :: pos_integer().
+-type interval() :: non_neg_integer().
+%% In milliseconds, defaults to 60000 (one minute) when not given.
+%% An interval of 0 means no delay at all, only the number of simultaneous executions will be
+%% controlled, which corresponds to the number of processes started
```
Comment (pawlooss1) [inline src/amoc_throttle/amoc_throttle.erl:27]: Same as for the previous comment I'd move this above the type it describes.

**Review by pawlooss1 — COMMENTED** [review]
All the docs look really nicer now 🔥 I left my additional thoughts on this.

File: src/amoc_controller.erl (hunk 1 above)
Comment (NelsonVides) [reply src/amoc_controller.erl:43]: The funny thing, is that it needs to be below. ex_doc will draw the line as a summary under the type if it comes _after_ the type 🥲 
Try building the docs locally to see 🤷🏽

File: src/amoc.erl (hunk 4 above)
Comment (NelsonVides) [reply src/amoc.erl:2]: That's for elixir, for erlang it doesn't work 🥲

File: src/amoc_controller.erl (hunk 1 above)
Comment (pawlooss1) [reply src/amoc_controller.erl:43]: Oh I see. Thank you for explaining this.

File: src/amoc_throttle/amoc_throttle.erl (hunk 5 above)
Comment (NelsonVides) [reply src/amoc_throttle/amoc_throttle.erl:27]: And again, needs to be this way, below, limitations of ex_doc 🥲

**Review by pawlooss1 — APPROVED** [review]
Thank you for the changes 🚀

**Pull request merged**